assets/app/scripts/services/routes.js

```diff
@@ -62,6 +62,42 @@ angular.module("openshiftConsole")
       }
     };
 
+    var addIngressWarnings = function(route, warnings) {
+      angular.forEach(route.status.ingress, function(ingress) {
+        var condition = _.find(ingress.conditions, { type: "Admitted", status: "False" });
+        if (condition) {
+          // TODO currently a controller bug where ingress host isn't getting set on failures
+          var message = 'Requested host ' + (ingress.host || route.spec.host) + ' was rejected by the router.';
+          if (condition.message || condition.reason) {
+            message += " Reason: " + (condition.message || condition.reason) + '.';
+          }
+          warnings.push(message);
+        }
+      });
+    };
+
+
+    function isCustomHost(route) {
+      return $filter('annotation')(route, "openshift.io/host.generated") !== "true";
+    }
+
+    // Gets the preferred route to display between two routes
+    // Preference order: admitted custom host with TLS -> admitted custom host -> custom host -> any route 
+    var getPreferredDisplayRoute = function(lhs, rhs) {
```

Comment:
You could assign scores and pick the highest score, but I'm fine with this, too.